v2/scheduler/internal/lib/queue/amqp/reader_test.go

```diff
@@ -0,0 +1,326 @@
+package amqp
+
+import (
+	"context"
+	"errors"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/Azure/go-amqp"
+	myamqp "github.com/brigadecore/brigade/v2/internal/amqp"
+	"github.com/brigadecore/brigade/v2/scheduler/internal/lib/queue"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetReaderFactoryConfig(t *testing.T) {
+	// Note that unit testing in Go does NOT clear environment variables between
+	// tests, which can sometimes be a pain, but it's fine here-- so each of these
+	// test cases builds on the previous case.
+	testCases := []struct {
+		name       string
+		setup      func()
+		assertions func(ReaderFactoryConfig, error)
+	}{
+		{
+			name:  "AMQP_ADDRESS not set",
+			setup: func() {},
+			assertions: func(_ ReaderFactoryConfig, err error) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "value not found for")
+				require.Contains(t, err.Error(), "AMQP_ADDRESS")
+			},
+		},
+		{
+			name: "AMQP_USERNAME not set",
+			setup: func() {
+				os.Setenv("AMQP_ADDRESS", "foo")
+			},
+			assertions: func(_ ReaderFactoryConfig, err error) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "value not found for")
+				require.Contains(t, err.Error(), "AMQP_USERNAME")
+			},
+		},
+		{
+			name: "AMQP_PASSWORD not set",
+			setup: func() {
+				os.Setenv("AMQP_USERNAME", "bar")
+			},
+			assertions: func(_ ReaderFactoryConfig, err error) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "value not found for")
+				require.Contains(t, err.Error(), "AMQP_PASSWORD")
+			},
+		},
+		{
+			name: "SUCCESS not set",
```

Comment:
nit: this test name is a bit confusing.  Maybe just "Success" or "GetReaderFactoryConfig Success"?

Reply:
Wow. I don't even know what happened there. I will straighten that out prior to merging.